lib/puppet_x/sqlserver/sql_connection.rb

```diff
@@ -9,7 +9,7 @@ def open_and_run_command(query, config)
           open(config)
           execute(query)
         rescue win32_exception => e
-          return ResultOutput.new(true, e.message)
+          return ResultOutput.new(true, @connection.Errors(0).Description)
```

Comment:
Am I correct in thinking this will only send back the first error? Is it possible/probable to get 2+ errors, and if so, do I care to retrieve the others?

Reply:
Yes, it will only send back the first error. It seems unlikely enough to receive multiple errors back from a script that it doesn't seem worth the increase in complexity to hand the edge case.

Reply:
To play devils advocate, also not too much to collect all the descriptions:

```ruby
messages = (0..@connection.Errors.Count { |i| @connection.Errors(i).Description }).join("\n")
```